### PR TITLE
Patches

### PR DESCRIPTION
**Pull request by sharkykh**
### Fixes:

* [These types of lines in the generated change log](https://github.com/SickRage/sickrage.github.io/commit/b11058aebcbffa6a23e3fe2807e1ae8440bee381#diff-b6264438a55b155f7c037101cc357ac9R5) (dunno why it started randomly showing).
[This **doesn't** fix all of them](https://github.com/SickRage/sickrage.github.io/commit/b11058aebcbffa6a23e3fe2807e1ae8440bee381#diff-b6264438a55b155f7c037101cc357ac9R27).
@miigotu  This shouldn't break anything for you. BTW not that it really matters because it's for you,
but the `$$$` delimiter fix has actually broke `genchanges` on my end for some reason.
This generates a different number for each Git Bash **instance**:
```bash
$ echo "\\$$\$"
\7356$
```
My guess is that Gruntfile.js is trying to split on "$$$" and can't find anything - 
'cause it returns `allTags` as an empty array. I fixed it locally by changing the delimiter on the git command, but didn't want to break it again for you after all the hard work you put in to make it work for you...
But if using `$$$` as a delimiter is what broke it for you, couldn't you just escape it? `\$\$\$`?
* Some QC minor issues:
  * Avoid consecutive if-statements
  * Prefer `format()` over string interpolation operator
* Empty massUpdate.js (I guess my previous commit didn't actually remove the file, just the contents)
* pylint error on six.moves

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

**Comment by miigotu** [comment]
If I dont escape that $ it only shows up with one $ in the date for me. This is because both $ and $$ are special variables in bash.

```bash
dusted@ROG:~/Projects/SickRage: echo $
$
dusted@ROG:~/Projects/SickRage: echo $$
7873
dusted@ROG:~/Projects/SickRage: echo $$$
7873$
dusted@ROG:~/Projects/SickRage: echo \$$$
$7873
dusted@ROG:~/Projects/SickRage: echo \$\$$
$$$
dusted@ROG:~/Projects/SickRage: echo \$\$\$
$$$
```

$$ is pid
```bash
dusted@ROG:~/Projects/SickRage: echo $$
7873
dusted@ROG:~/Projects/SickRage: ps aux | grep "/bin/bash --login"
dusted    7873  0.0  0.0  26264  8712 pts/0    Ss   21:30   0:00 /bin/bash --login
```

**Comment by sharkykh** [comment]
Then what you're saying is we can escape it and it will still work for you, then it will work for me too.
I know Git Bash isn't real bash but still:
```bash
Kfir@Kfir-Laptop MINGW64 ~
$ echo $
$

Kfir@Kfir-Laptop MINGW64 ~
$ echo $$
1176

Kfir@Kfir-Laptop MINGW64 ~
$ echo $$$
1176$

Kfir@Kfir-Laptop MINGW64 ~
$ echo \\$$$
\1176$

Kfir@Kfir-Laptop MINGW64 ~
$ echo \$$$
$1176

Kfir@Kfir-Laptop MINGW64 ~
$ echo \$\$$
$$$

Kfir@Kfir-Laptop MINGW64 ~
$ echo \$\$\$
$$$
```

**Comment by miigotu** [comment]
We will see, for some reason \$\$\$ didnt work from inside grunt for me, but maybe I had a brain fart moment and made a bad test. Adjust it and we will see =P

**Comment by sharkykh** [comment]
lol happend to me too
```bash
Kfir@Kfir-Laptop MINGW64 /C/Dev/sickrage (patches)
$ grunt git_list_tags
Warning: Task "git_list_tags" not found. Use --force to continue.

Aborted due to warnings.

Kfir@Kfir-Laptop MINGW64 /C/Dev/sickrage (patches)
$ grunt exec:git_list_tags
Running "exec:git_list_tags" (exec) task
>> [object Object],[object Object],[object Object],...........
Done.
```

BTW,
👍 👍 👍  on AppVeyor 😄

**Comment by sharkykh** [comment]
If you want add me or the Github team to the AppVeyor project, maybe I can help?

**Comment by sharkykh** [comment]
I have a working tox.ini for windows

**Comment by sharkykh** [comment]
I have a working tox.ini for windows
[tox_win.txt](https://github.com/SickRage/SickRage/files/857156/tox_win.txt)

**Comment by miigotu** [comment]
Im trying to figure out if there is a way to adjust the rm -rf when on windows without specifying an alternate tox.ini
del /f /s /q blah.db

**Comment by sharkykh** [comment]
oh

**Comment by sharkykh** [comment]
https://github.com/sashkab/ttt/blob/master/tox.ini
?

hmm maybe I'll dive into the docs before suggesting stuff 😆

**Comment by miigotu** [comment]
Its a PITA =P

**Comment by sharkykh** [comment]
🤣 You said pita and I instantly thought about [the food](https://en.wikipedia.org/wiki/Pita).

**Comment by sharkykh** [comment]
The reason it didn't work is this: http://tox.readthedocs.io/en/latest/config-v2.html (scroll down to the bottom)

MAYBE this will work?
but I see you gave up haha
```
commands =
    linux: /bin/rm -f {toxinidir}/tests/sickbeard.db \
		/bin/rm -f {toxinidir}/tests/cache.db \
		/bin/rm -f {toxinidir}/tests/failed.db \
		nosetests -c nose.cfg --nocapture \
		- coveralls
    win: del /f /s /q {toxinidir}/tests/sickbeard.db \
		del /f /s /q {toxinidir}/tests/cache.db \
		del /f /s /q {toxinidir}/tests/failed.db \
		nosetests -c nose.cfg --nocapture
		
```

**Comment by miigotu** [comment]
Ah only one line allowed for each one? I think they can subcommand:
```
commands=
    linux:
        - blah
        - blah2
    win:
        - ok
        - ok
```

**Comment by sharkykh** [comment]
Why are we testing on AppVeyor when I'm on windows :grin:
Hold on

**Comment by miigotu** [comment]
All that time I was pushing to develop by accident -.-

**Comment by miigotu** [comment]
@sharkykh 
https://ci.appveyor.com/project/miigotu/sickrage/build/306/job/l3f908f3500kkodw#L2899

**Comment by sharkykh** [comment]
Yeah I noticed, but we're supposed to be able to make tox cross platform... Still on it.

**Comment by sharkykh** [comment]
I'm close :)
```
C:\Dev\SickRage>tox -v -e py27-win
using tox.ini: C:\Dev\SickRage\tox.ini
using tox-2.6.0 from c:\python27\lib\site-packages\tox\__init__.pyc
py27-win reusing: C:\Dev\SickRage\.tox\py27-win
  C:\Dev\SickRage$ C:\Dev\SickRage\.tox\py27-win\Scripts\pip.EXE freeze >C:\Dev\SickRage\.tox\py27-win\log\py27-win-18.log
py27-win installed: appdirs==1.4.3,packaging==16.8,pyparsing==2.2.0,six==1.10.0
py27-win runtests: PYTHONHASHSEED='644'
py27-win runtests: commands[0] | cmd /c del /f /q C:\Dev\SickRage\tests\sickbeard.db
WARNING:test command found but not installed in testenv
  cmd: C:\Windows\system32\cmd.EXE
  env: C:\Dev\SickRage\.tox\py27-win
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
  C:\Dev\SickRage$ C:\Windows\system32\cmd.EXE /c del /f /q C:\Dev\SickRage\tests\sickbeard.db
Could Not Find C:\Dev\SickRage\tests\sickbeard.db
py27-win runtests: commands[1] | cmd /c del /f /q C:\Dev\SickRage\tests\cache.db
WARNING:test command found but not installed in testenv
  cmd: C:\Windows\system32\cmd.EXE
  env: C:\Dev\SickRage\.tox\py27-win
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
  C:\Dev\SickRage$ C:\Windows\system32\cmd.EXE /c del /f /q C:\Dev\SickRage\tests\cache.db
Could Not Find C:\Dev\SickRage\tests\cache.db
py27-win runtests: commands[2] | cmd /c del /f /q C:\Dev\SickRage\tests\failed.db
WARNING:test command found but not installed in testenv
  cmd: C:\Windows\system32\cmd.EXE
  env: C:\Dev\SickRage\.tox\py27-win
Maybe you forgot to specify a dependency? See also the whitelist_externals envconfig setting.
  C:\Dev\SickRage$ C:\Windows\system32\cmd.EXE /c del /f /q C:\Dev\SickRage\tests\failed.db
Could Not Find C:\Dev\SickRage\tests\failed.db
___________________________________ summary ___________________________________
  py27-win: commands succeeded
  congratulations :)

```
disabled --recreate and nosetests just so it doesn't run while i'm configuring tox. but they work

**Comment by miigotu** [comment]
whitelist_externals = /bin/rm, cmd

**Comment by sharkykh** [comment]
nope, even when it's set to all (`whitelist_externals = *`) [tox configuration](http://tox.readthedocs.io/en/latest/config.html#confval-whitelist_externals=MULTI-LINE-LIST)

current:
```ini
[tox]
skipsdist = true
platform=
    win: windows
    linux: linux
envlist = py27-{win,linux}
whitelist_externals = *
#    /bin/rm
#	%windir%\system32\cmd.exe

[testenv:py27-win]
envdir = {toxworkdir}/tox
deps =
    coveralls
    nose
    rednose
    mock
    vcrpy-unittest
    pytz
#needs to be appveyor:
#passenv = TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH
commands =
    cmd /c del /f /q {toxinidir}\tests\sickbeard.db
	cmd /c del /f /q {toxinidir}\tests\cache.db
	cmd /c del /f /q {toxinidir}\tests\failed.db
	#nosetests -c nose.cfg --nocapture

[testenv:py27-linux]
envdir = {toxworkdir}/tox
deps =
    coveralls
    nose
    rednose
    mock
    vcrpy-unittest
    pytz
passenv = TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH
commands =
    /bin/rm -f {toxinidir}/tests/sickbeard.db
	/bin/rm -f {toxinidir}/tests/cache.db
	/bin/rm -f {toxinidir}/tests/failed.db
	nosetests -c nose.cfg --nocapture
	- coveralls
```

**Comment by miigotu** [comment]
Its nbd, an extra tox-win.ini is not a major issue. Its working atm.

**Comment by sharkykh** [comment]
Yeah it still bugs me. I will make a PR if I find a better solution :)

**Comment by miigotu** [comment]
Does `unlink c:\projects\SickRage\sickbeard.db` work in windows?
Note: *.db is not in cwd, it's in c:\projects\SickRage
Oh yeah, you dont need to delete the files really anyways, that is just to make testing locally easier.

**Comment by sharkykh** [comment]
no such command

**Comment by miigotu** [comment]
Or you could add a grunt task that cleans up Logs, cache/, *.db ^^ and call that from tox.ini

**Comment by sharkykh** [comment]
That's actually a good idea, but still. If it doesn't recognize cmd.exe it's a big issue for the future windows tests

**Comment by sharkykh** [comment]
Damn what a moron I am. look in the txt I uploaded like 10 comments ago while you were still testing.
look at how I solved this very same issue.
**TL;DR got that working now**
just need to add some kind of tox var to use the users/CI %windir% env var

**Comment by sharkykh** [comment]
Don't merge yet,
I am gonna solve the issue with windows paths first.

**Comment by sharkykh** [comment]
I broke it with the regex in indexOf. looking for an alternative

**Comment by sharkykh** [comment]
I'm done.

**Comment by miigotu** [comment]
```javascript
file.substr(0, Math.max(file.lastIndexOf('/'), file.lastIndexOf('\\')))
```
or
```javascript
file.substr(0, file.lastIndexOf('CHANGES.md'))
```

**Comment by miigotu** [comment]
Hard slice works too I guess

**Comment by sharkykh** [comment]
you can't really mess up like that
only if you change the location of CHANGES.md on the repo..

**Comment by miigotu** [comment]
Ugh, you broke it for me haha
```
Running "exec:git_get_last_tag" (exec) task

Running "exec:git_list_tags" (exec) task

Running "_genchanges" task
Fatal error: No tags information was received.
```

**Comment by miigotu** [comment]
Changed it to @@@ to save us headache

**Comment by sharkykh** [comment]
What the hell is wrong with it.

**Comment by sharkykh** [comment]
Actually I know what. I don't think it's supposed to be used that way haha

**Comment by miigotu** [comment]
```k icons\n9528$\n' ]```

Anyways, still have a problem now that I changed that....

**Comment by sharkykh** [comment]
What was the issue before? I thought `\$\$\$` was supposed to work for you.

**Comment by miigotu** [comment]
No, I had to double escape the first $ because it is both $ $$ and $$ $, and either $$ will return the pid.

**Comment by sharkykh** [comment]
Lol we could always use like AAAAAAAAAAA as a delimiter

**Comment by miigotu** [comment]
I figured @@@ would never be used in context, so I think its ok

**Comment by sharkykh** [comment]
Great-- Latest develop works on Git Bash AND cmd.
👍
